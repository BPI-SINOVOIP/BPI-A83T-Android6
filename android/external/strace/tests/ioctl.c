#ifdef HAVE_CONFIG_H
# include "config.h"
#endif
#include <fcntl.h>
#include <stdint.h>
#include <unistd.h>
#include <termios.h>
#include <sys/ioctl.h>

#ifdef HAVE_LINUX_MMTIMER_H
# include <linux/mmtimer.h>
#endif
#ifdef HAVE_LINUX_HIDDEV_H
# include <linux/hiddev.h>
#endif
#ifdef HAVE_LINUX_INPUT_H
# include <linux/input.h>
#endif

#if defined MMTIMER_GETRES \
 && defined HIDIOCGVERSION \
 && defined HIDIOCGPHYS \
 && defined EVIOCGBIT \
 && defined EV_KEY

int
main(void )
{
	struct termios tty;
	uint64_t data;

	if (ioctl(-1, TCGETS, &tty) != -1 ||
	    ioctl(-1, MMTIMER_GETRES, &data) != -1 ||
	    ioctl(-1, HIDIOCGVERSION, &data) != -1 ||
	    ioctl(-1, HIDIOCGPHYS(8), &data) != -1 ||
	    ioctl(-1, EVIOCGBIT(EV_KEY, 8), &data) != -1 ||
	    ioctl(-1, _IOR(0xde, 0xad, data), &data) != -1)
		return 77;

	return 0;
}

#else

int
main(void )
{
	return 77;
}

#endif
