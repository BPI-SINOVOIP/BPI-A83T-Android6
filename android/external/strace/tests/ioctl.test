#!/bin/sh

# Check ioctl syscall decoding.

. "${srcdir=.}/init.sh"

run_prog
run_strace -e ioctl $args
match_grep

exit 0
