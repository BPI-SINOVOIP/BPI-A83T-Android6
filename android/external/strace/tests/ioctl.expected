ioctl\(-1, TCGETS, 0x[0-9a-f]+\) += -1 EBADF .*
ioctl\(-1, MMTIMER_GETRES, 0x[0-9a-f]+\) += -1 EBADF .*
ioctl\(-1, HIDIOCGRDESCSIZE or HIDIOCGVERSION, 0x[0-9a-f]+\) += -1 EBADF .*
ioctl\(-1, HIDIOCGPHYS\(8\), 0x[0-9a-f]+\) += -1 EBADF .*
ioctl\(-1, EVIOCGBIT\(EV_KEY, 8\), 0x[0-9a-f]+\) += -1 EBADF .*
ioctl\(-1, _IOC\(_IOC_READ, 0xde, 0xad, 0x08\), 0x[0-9a-f]+\) += -1 EBADF .*
